### PR TITLE
Allow overriding kafka host for clickhouse via KAFKA_URL_CLICKHOUSE env var

### PR DESCRIPTION
**Pull request by macobo**
This is needed when using an external clickhouse which doesn't have the
same access to kafka as in-cluster traffic does.

Note that long-term we might need to also provide better auth mechanisms
here as well.

Related: https://github.com/PostHog/charts-clickhouse/pull/276

### CODE REVIEW

File: posthog/settings/data_stores.py [hunk 1]
```diff
@@ -101,9 +101,16 @@
...
+_parse_kafka_hosts = lambda kafka_url: ",".join(urlparse(host).netloc for host in kafka_url.split(","))
+
+# URL Used by kafka clients/producers
 KAFKA_URL = os.getenv("KAFKA_URL", "kafka://kafka")
-KAFKA_HOSTS_LIST = [urlparse(host).netloc for host in KAFKA_URL.split(",")]
-KAFKA_HOSTS = ",".join(KAFKA_HOSTS_LIST)
+KAFKA_HOSTS = _parse_kafka_hosts(KAFKA_URL)
+
+# Kafka broker host(s) that is used by clickhouse for ingesting messages. Useful if clickhouse is hosted outside the cluster.
+KAFKA_HOSTS_CLICKHOUSE = _parse_kafka_hosts(os.getenv("KAFKA_URL_CLICKHOUSE", KAFKA_URL))
```
Comment (guidoiaquinti) [inline posthog/settings/data_stores.py:112]: What about `KAFKA_URL_EXTERNAL`?

Comment (macobo) [reply posthog/settings/data_stores.py:112]: Not used anywhere hence thought best not to expose it.

Comment (macobo) [reply posthog/settings/data_stores.py:112]: From discussion: `KAFKA_HOSTS_FOR_CLICKHOUSE` and similar name for the env var :)

**Review by guidoiaquinti — APPROVED** [review]
CI is failing but overall LGTM

**Pull request merged**